pkg/minikube/machine/cache_images.go

```diff
@@ -314,10 +314,19 @@ func CacheImage(image, dst string) error {
 	}
 
 	glog.Infoln("OPENING: ", dstPath)
-	f, err := os.Create(dstPath)
+	tmpPath := dstPath + ".tmp"
+	f, err := os.Create(tmpPath)
 	if err != nil {
 		return err
 	}
-	defer f.Close()
-	return tarball.Write(tag, img, nil, f)
+	err = tarball.Write(tag, img, nil, f)
+	if err != nil {
+		return err
+	}
+	f.Close()
```

Comment:
Please check the error code to close!

Reply:
I think we did it with `defer` before... WIll check